daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -186,21 +187,170 @@ private[lf] object Speedy {
 
     private[this] var contractsCache = Map.empty[V.ContractId, V.ContractInstance]
 
-    def lookupContractOnLedger[Q](coid: V.ContractId)(
+    // To handle continuation exceptions, as continuation are run outside the interpreter loop.
+    private[this] def safelyContinue(
```

Comment:
```suggestion
    // To handle continuation exceptions, as continuations run outside the interpreter loop.
    // Here we delay the throw to the interpreter loop, but it would be probably better 
    // to delay the whole execution. This would work for all continuation (which have type
    // `X => Unit`)  except `NeedKey` (which have type `X => Bool`) that need to be run
    // strait away.  
    private[this] def safelyContinue(
```